packages/repo-tools/src/commands/knip-reports/knip-extractor.ts

```diff
@@ -55,41 +58,43 @@ function logKnipReportInstructions() {
   console.log('');
 }
 
-async function generateKnipConfig({ packageDir }: KnipConfigOptions) {
-  const knipConfig = {
-    entry: [
-      'dev/index.{ts,tsx}',
-      'src/index.{ts,tsx}',
-      'src/alpha.{ts,tsx}',
-      'src/routes.ts',
-      'src/run.ts',
-    ],
+async function generateKnipConfig({ knipConfigPath }: KnipConfigOptions) {
+  const knipConfig: KnipConfig = {
+    workspaces: {
+      '.': {},
+      '{packages,plugins}/*': {
+        entry: ['dev/index.{ts,tsx}', 'src/index.{ts,tsx}'],
+        ignore: [
+          '.eslintrc.js',
+          'config.d.ts',
+          'knexfile.js',
+          'node_modules/**',
+          'dist/**',
+          '{fixtures,migrations,templates}/**',
+          'src/tests/transforms/__fixtures__/**', // cli packaging tests
+        ],
+      },
+    },
     jest: {
-      entry: ['src/setupTests.ts', '**/*.test.{ts,tsx}'],
+      entry: ['src/setupTests.ts', 'src/**/*.test.{ts,tsx}'],
     },
     storybook: { entry: 'src/components/**/*.stories.tsx' },
-    ignore: [
-      '.eslintrc.js',
-      'config.d.ts',
-      'knexfile.js',
-      'node_modules/**',
-      'dist/**',
-      '{fixtures,migrations,templates}/**',
-    ],
     ignoreDependencies: [
+      // these is reported as a referenced optional peerDependencies
+      // TBD: investigate what triggers these
+      '@types/react',
+      '@types/jest',
+      '@internal/.*', // internal packages are not published and inlined
```

Comment:
The trouble with this is, `@internal` is a very common prefix and not always meaning it isn't published. Actually I don't know if the package name can be used for anything truly useful here 🤔 But yeah, given that his feature really isn't truly fully usable for external people yet (that would require some more configurability etc probably), this is still true for our own repo and some common cases, so let's keep it as is for now.